pym/bob/generators/VisualStudio.py

```diff
@@ -0,0 +1,320 @@
+      <Extensions>cpp;c;cxx;def;odl;idl;hpj;bat;asm;asmx</Extensions>
+    </Filter>
+    <Filter Include="Header Files">
+      <UniqueIdentifier>{{93995380-89BD-4b04-88EB-625FBE52EBFB}}</UniqueIdentifier>
+      <Extensions>h;hpp;hxx;hm;inl;inc;xsd</Extensions>
+    </Filter>
+    <Filter Include="Resource Files">
+      <UniqueIdentifier>{{67DA6AB6-F800-4c08-8B7A-83BB121AAD01}}</UniqueIdentifier>
+      <Extensions>rc;ico;cur;bmp;dlg;rc2;rct;bin;rgs;gif;jpg;jpeg;jpe;resx</Extensions>
+    </Filter>
+  </ItemGroup>
+{FILTERS}
+</Project>
+"""
+
+#def relpath(pathFrom, pathTo):
+#    if pathTo.is_absolute():
+#        return str(pathTo)
+#
+#    if pathFrom.is_absolute():
+#        ret = [ pathFrom.drive ]
+#        pathFrom = list(pathFrom.parts)[1:]
+#    else:
+#        ret = []
+#        pathFrom = list(pathFrom.parts)
+#    pathTo = list(pathTo.parts)
+#
+#    while pathFrom and not 
+#
+#    return 
+
+class Project:
+    def __init__(self, recipesRoot, uuid, packagePath):
+        self.uuid = uuid
+        self.recipesRoot = recipesRoot
+        self.packagePath = packagePath
+        self.headers = []
+        self.sources = []
+        self.resources = []
+        self.incPaths = []
+        self.dependencies = set()
+
+    def addHeader(self, header):
+        #self.headers.append(str(header.relative_to(self.root)))
+        self.headers.append(str(self.recipesRoot.joinpath(header)))
+    def addSource(self, source):
+        #self.sources.append(str(source.relative_to(self.root)))
+        self.sources.append(str(self.recipesRoot.joinpath(source)))
+    def addResource(self, resource):
+        #self.resources.append(str(resource.relative_to(self.root)))
+        self.resources.append(str(self.recipesRoot.joinpath(resource)))
+    def addIncPath(self, path):
+        self.incPaths.append(str(self.recipesRoot.joinpath(path)))
+    def addDependency(self, dep):
+        self.dependencies.add(dep)
+
+    def generateProject(self, projects, cmd):
+        items = []
+        if self.headers:
+            g = ElementTree.Element("ItemGroup")
+            for i in self.headers: ElementTree.SubElement(g, "ClInclude", {"Include" : str(i)})
+            items.append(ElementTree.tostring(g, encoding="unicode"))
+        if self.sources:
+            g = ElementTree.Element("ItemGroup")
+            for i in self.sources: ElementTree.SubElement(g, "ClCompile", {"Include" : str(i)})
+            items.append(ElementTree.tostring(g, encoding="unicode"))
+        # TODO: self.resources
+        if self.dependencies:
+            g = ElementTree.Element("ItemGroup")
+            incPaths = []
+            for i in sorted(self.dependencies):
+                dep = ElementTree.SubElement(g, "ProjectReference",
+                        {"Include" : "..\\{NAME}\\{NAME}.vcxproj".format(NAME=i)})
+                ElementTree.SubElement(dep, "Project").text = "{" + str(projects[i].uuid) + "}"
```

Comment:
If there is a module, which is splitted into multiple packages, the projects[i] will crash.
As workaround, I surrounded this lines by try-except and everything is fine.

Reply:
fixed!